docs/tutorials/python/sagas/index.md

```diff
@@ -0,0 +1,693 @@
+---
+id: sagas
+sidebar_position: 4
+keywords: [Python, temporal, sdk, tutorial, saga, saga pattern, rollback, compensating actions]
+tags: [Python, SDK]
+last_update:
+  date: 2024-03-06
+title: Build a trip booking application in Python
+description: Implement the Saga Pattern in Python using Temporal.
+image: /img/temporal-logo-twitter-card.png
+---
+
+When dealing with distributed systems, a failure in one service can lead to a domino effect, compromising the entire transaction.
+The Saga pattern offers a solution to this problem by allowing distributed transactions to be broken into smaller, manageable transactions, each with its own compensation logic in case of failure.
+
+The Saga pattern is a design pattern that provides a mechanism to manage long-running transactions and ensure data consistency across multiple services.
+Instead of a single monolithic transaction, the Saga pattern breaks the transaction into smaller, manageable steps (Activities), each step is executed sequentially, and if a step fails, previous steps are undone with a compensating step.
+
+Temporal orchestrates long-running transactions, automatically compensating for failures.
+The compensation, combined with the guarantee that the method will complete execution, makes this method a reliable, long-running transaction.
+
+With this guide, you'll build a Flask API that uses Temporal to manage the booking process for cars, hotels, and flights.
+This approach ensures that even if one part of the booking fails, the system can gracefully handle the rollback of previous steps, maintaining data consistency.
+
+When you're finished, you'll be able to handle complex distributed transactions with ease and reliability using Temporal.
+
+## Prerequisites
+
+Before you begin, ensure you have the following:
+
+- Complete the [Hello World](/getting_started/python/hello_world_in_python/index.md) tutorial
+- Familiarity with asynchronous programming in Python
+- Basic understanding of microservices and distributed systems
+- The [Flask framework with async extras](https://flask.palletsprojects.com/en/2.3.x/async-await/) installed as a dependency of your project, which you can do with `pip install flask[async]`.
+
+## Create the booking functions
+
+You will start by creating the Activities for the booking process.
```

Comment:
I think there's a little bit missing here. If you assume no knowledge of temporal and you land here, you might want to have something here that explains that you're going to use temporal workflows and activities to build this logic. 


```suggestion
You will start by creating the individual components for the booking process. You'll create a Temporal Workflow and a set of Activities. Activities let you interact with services, and Workflows orchestrate the Activities.
```

And I'd expect both workflows and activities to be linked to appropriate encyclopedia pages in docs. I did this in the next.js tutorial. We aren't asking for much, but this one sentence can help explain how the functions map to the core features.